### PR TITLE
Small refactor of the bridge plugin

### PR DESCRIPTION
**Pull request by almusil**
Refactor the attachPods function and make it parallel
without the need of list.

Signed-off-by: Ales Musil <amusil@redhat.com>

### CODE REVIEW

**Comment by almusil** [comment]
@phoracek @yuvalif

File: pkg/network/bridge/plugin.go [hunk 1]
```diff
@@ -189,53 +187,48 @@ func getAssignmentPath(bridge string, nic string) string {
...
+						return false
+					}
+
+					containerPid, err := cli.GetPidByContainerID(containerID)
+					if err != nil {
+						glog.V(3).Info("Failed to obtain container's pid, due to: %s", err.Error())
+						return false
+					}
+
+					attachMutex.Lock()
```
Comment (yuvalif) [inline pkg/network/bridge/plugin.go:215]: why is the mutex needed? is it because of the netns.Set() call?

Comment (phoracek) [reply pkg/network/bridge/plugin.go:215]: I think so, I feel it would introduce some race problems without the lock.

**Review by phoracek — APPROVED** [review]

**Comment by phoracek** [comment]
LGTM, waiting for @yuvalif ack

File: pkg/network/bridge/plugin.go (hunk 1 above)
Comment (yuvalif) [reply pkg/network/bridge/plugin.go:215]: ok

**Review by yuvalif — APPROVED** [review]

**Comment by phoracek** [comment]
Thanks @almusil, good work

**Pull request merged**